virttest/libvirt_remote.py

```diff
@@ -0,0 +1,46 @@
+"""
+Shared code for tests that need to control libvirt on remote
+"""
+
+import logging
+
+from avocado.core import exceptions
+
+from virttest import remote
+
+from virttest.utils_test import libvirt
+
+
+def update_remote_file(params, new_key,
+                       file_path='/etc/libvirt/libvirtd.conf',
+                       restart_libvirt=True):
+    """
+    Update file on remote and restart libvirtd if needed
+
+    :param param: Test run params
+    :param new_key: Keys to update
+    :param file_path: The path of the file
+    :param restart_libvirt: True to restart libvirtd
+    :return: remote.RemoteFile object
+
+    """
+    try:
+        tmp_key = eval(new_key)
+        logging.debug("Update file {} with: {}".format(file_path, new_key))
+        remote_ip = params.get("server_ip", params.get("remote_ip"))
+        remote_pwd = params.get("server_pwd", params.get("remote_pwd"))
+        remote_user = params.get("server_user", params.get("remote_user"))
+
+        remote_file_obj = remote.RemoteFile(address=remote_ip, client='scp',
+                                            username=remote_user,
+                                            password=remote_pwd,
+                                            port='22', remote_path=file_path)
+        remote_file_obj.sub_else_add(tmp_key)
+        if restart_libvirt:
+            libvirt.remotely_control_libvirtd(remote_ip, remote_user,
+                                              remote_pwd, action='restart',
+                                              status_error='no')
+        return remote_file_obj
```

Comment:
I assume there is no much meaning to return remote_file_obj

Reply:
@chunfuwen  When RemoteFile object deleted, the updated file was recovered. So I return it here.